pkg/workload/workload.go

```diff
@@ -237,3 +237,31 @@ func UpdateStatusIfChanged(ctx context.Context,
 	// Updating an existing condition
 	return UpdateStatus(ctx, c, wl, conditionType, conditionStatus, reason, message)
 }
+
+// AdmissionFrom returns only the fields necessary for admission using
+// ServerSideApply.
+func AdmissionFrom(w *kueue.Workload) *kueue.Workload {
+	var admission *kueue.Admission
+	if w.Spec.Admission != nil {
+		admission = w.Spec.Admission.DeepCopy()
+	}
+	wlCopy := &kueue.Workload{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:        w.UID,
+			Name:       w.Name,
+			Namespace:  w.Namespace,
+			Generation: w.Generation, // Produce a conflict if there was a change in the spec.
+		},
+		TypeMeta: w.TypeMeta,
+		Spec: kueue.WorkloadSpec{
+			Admission: admission,
```

Comment:
I'm not convinced that setting `Admission: nil` works here. I think it wouldn't be serialized, because it's set to `omitempty`. We might need to create a specialized struct?

Reply:
We have integration tests to confirm that it works :) I think in SSA if you own a field (and `admission` is managed by `kueue-admission`), then if you don't send it back it is deleted.

Reply:
This is nice!